### PR TITLE
feat(DataTable): Add support for nested object in sort

### LINKED ISSUE #4484: DataTable - Ability to sort on a column.property which references a nested object value
### Expected Behavior

If I have a data set in the shape of:
```
[
    {
        name: 'Johannes',
        favorites: {
            color: 'blue',
            movie: 'Butch Cassidy and the Sundance Kid',
            number: 3,
        },
    },
    ...
]
```

I would like the ability to sort on a column.property which references a nested object value.

### Actual Behavior

When
```
<DataTable data={data} columns={columns} sortable />
```
And `columns` contains
```
[
    ...
    {
        property: 'favorites.number'
    }
    ...
]
```
the DataTable renders the values, but the column is not sortable.

### URL, screen shot, or Codepen exhibiting the issue

https://codesandbox.io/s/jolly-cookies-l5x15?file=/src/App.js

[Related Issue](https://github.com/hpe-design/design-system/issues/1079)

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version: 2.15 / Stable
- Browser Name and version:
- Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by netochaves**
Closes #4484
#### What does this PR do?
Adds support for nested object in sort of DataTable
#### Where should the reviewer start?
`src/js/components/DataTable/buildState.js `
#### What testing has been done on this PR?
 Unit tests
#### How should this be manually tested?
Storybook
#### What are the relevant issues?
#4484
#### Do the grommet docs need to be updated?
I don't think so
#### Should this PR be mentioned in the release notes?
I don't think so
[Shimi] Yes, DataTable to support nested object on sort

#### Is this change backwards compatible or is it a breaking change?
 It's backwards compatible

### CODE REVIEW

File: src/js/components/DataTable/__tests__/DataTable-test.js [hunk 1]
```diff
@@ -182,6 +182,42 @@ describe('DataTable', () => {
...
+              header: 'Value',
+              render: datum => datum.b && datum.b.value,
+            },
+          ]}
+          data={[
+            { a: 'zero', b: { value: 1 } },
+            { a: 'one', b: { value: 2 } },
+            { a: 'two', b: { value: 3 } },
+          ]}
+          sortable
```
Comment (ShimiSun) [inline src/js/components/DataTable/__tests__/DataTable-test.js:202]: If you don't use sortable, and just use onSort, would that still work? can we add a test for that as well?

**Review by ShimiSun — COMMENTED** [review]
Thank you for this PR! I dropped a small question.

File: src/js/components/DataTable/__tests__/DataTable-test.js (hunk 1 above)
Comment (netochaves) [reply src/js/components/DataTable/__tests__/DataTable-test.js:202]: Yeah, it will work, added a test for that.

**Comment by taysea** [comment]
Linking to https://github.com/hpe-design/design-system/issues/1079

**Review by halocline — APPROVED** [review]
This looks good to me. Thank you @netochaves. 👏

**Review by ericsoderberghp — APPROVED** [review]

**Comment by ericsoderberghp** [comment]
Thanks for your contribution!

**Pull request merged**